nautobot/ipam/models.py

```diff
@@ -116,29 +117,43 @@ class VRF(PrimaryModel):
         through="ipam.VRFDeviceAssignment",
         through_fields=("vrf", "device"),
     )
-    prefixes = models.ManyToManyField(
-        to="ipam.Prefix",
+    virtual_machines = models.ManyToManyField(
+        to="virtualization.VirtualMachine",
         related_name="vrfs",
-        through="ipam.VRFPrefixAssignment",
+        through="ipam.VRFDeviceAssignment",
+        through_fields=("vrf", "virtual_machine"),
     )
+    # prefixes = models.ManyToManyField(
+    #     to="ipam.Prefix",
+    #     related_name="vrfs",
+    #     through="ipam.VRFPrefixAssignment",
+    # )
     tenant = models.ForeignKey(
         to="tenancy.Tenant",
         on_delete=models.PROTECT,
         related_name="vrfs",
         blank=True,
         null=True,
     )
+    enforce_unique = models.BooleanField(
+        default=True,
+        verbose_name="Enforce unique space",
+        help_text="Prevent duplicate prefixes/IP addresses within this VRF",
+    )
     description = models.CharField(max_length=200, blank=True)
     import_targets = models.ManyToManyField(to="ipam.RouteTarget", related_name="importing_vrfs", blank=True)
     export_targets = models.ManyToManyField(to="ipam.RouteTarget", related_name="exporting_vrfs", blank=True)
 
-    csv_headers = ["name", "rd", "tenant", "description"]
+    # csv_headers = ["name", "rd", "tenant", "description"]
+    csv_headers = ["name", "rd", "tenant", "enforce_unique", "description"]
```

Comment:
Need to get latest `next` merged up (#3722) to the prototype branch and address merge conflicts... whee!